index.js

```diff
@@ -81,13 +92,21 @@ function compressPlugin (fastify, opts, next) {
       return next()
     }
 
+    var stream
+
     if (req.headers['x-no-compression'] !== undefined) {
-      return next()
+      payload = payloadAsStream(payload)
+      stream = maybeUnzip()
+      pump(payload, stream, onEnd.bind(reply))
+      return next(null, stream)
```

Comment:
I would prefer if we did _not_ convert it to a Stream if we do not have to unzip.

Reply:
Not a problem. If we extract out the `unzip-if-zip` logic into a standalone function, we can call it if the payload is a string or buffer.

Reply:
go for it!